cpu/nrf5x_common/periph/gpio_ll.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2015 Jan Wagner <mail@jwagner.eu>
+ *               2015-2016 Freie Universität Berlin
+ *               2019 Inria
+ *               2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_nrf5x_common
+ * @ingroup     drivers_periph_gpio_ll
+ * @{
+ *
+ * @file
+ * @brief       Peripheral GPIO Low-Level API implementation for the nRF5x MCU family
+ *
+ * @author      Christian Kühling <kuehling@zedat.fu-berlin.de>
+ * @author      Timo Ziegler <timo.ziegler@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Jan Wagner <mail@jwagner.eu>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+
+#include "cpu.h"
+#include "periph/gpio_ll.h"
+#include "periph_cpu.h"
+#include "periph_conf.h"
+
+int gpio_ll_init(gpio_port_t port, uint8_t pin, gpio_conf_t *conf)
```

Comment:
Looks like I forgot to apply the pull resistor config. I also recall that the test for the pull resistors did pass right away for the STM32, but had to be "tweaked" to allow some time to pass until the pull resistor managed to pull the pin to the desired level. Apparently that tweak made the test pass even with a floating input, as the noise it will read eventually has the expected logic level :(

I guess it is time to revert that tweak and implement the pull resistor configuration...